frontend/public/components/secrets/create-secret/PullSecretUploadForm.tsx

```diff
@@ -0,0 +1,54 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { DroppableFileInput, PullSecretData } from '.';
+
+export const PullSecretUploadForm: React.FC<PullSecretUploadFormProps> = ({
+  onChange,
+  onDisable,
+  stringData,
+}) => {
+  const { t } = useTranslation();
+  const [configFile, setConfigFile] = React.useState<string>(
+    _.isEmpty(stringData) ? '' : JSON.stringify(stringData),
+  );
+  const [parseError, setParseError] = React.useState<boolean>(false);
+
+  const onFileChange = React.useCallback(
+    (fileData: string) => {
+      const parsedData = _.attempt(JSON.parse, fileData);
+      setConfigFile(fileData);
+      setParseError(_.isError(parsedData));
+      onChange(parsedData);
+      onDisable(_.isError(parsedData));
```

Comment:
Prefer determinate try/catch rather than _.attempt:
```suggestion
     try {
       const newPullSecret = JSON.parse(fileData);
       onChange(newPullSecret);
       setParseError(false);
       onDisable(false);
     } catch (e) {
       setParseError(true);
       onDisable(true);
     }
```